app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt

```diff
@@ -76,6 +76,11 @@ class EditProfileFragment : Fragment() {
 
         editProfileViewModel.message.observe(this, Observer {
             Toast.makeText(context, it, Toast.LENGTH_LONG).show()
+            editProfileViewModel.isUpdated.observe(this, Observer { isUpdated ->
+                isUpdated?.let {
```

Comment:
> this can be shortened, see how it is being done in other places in the file. use `it`

But in that case when there is an error in updating the profile, then also it direct to profile activity. Is that okay

Reply:
also check if `isUpdated` is true. This should not happen

Reply:
> also check if `isUpdated` is true. This should not happen

Sorry, I don't get it because for isUpdated also need to observe the boolean then the code becomes same. And I already use let operator instead of if.

Reply:
Inside `it?.let {`
write `if (it) { ...... }`

Reply:
Updated, now it's working fine without if statement.

Reply:
I meant you could write it how it is being done here
https://github.com/fossasia/open-event-android/blob/192fb005917a0cbd74ddddcff797b7c6848140ae/app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt#L88

Reply:
> I meant you could write it how it is being done here
> [open-event-android/app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt](https://github.com/fossasia/open-event-android/blob/192fb005917a0cbd74ddddcff797b7c6848140ae/app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt#L88)
> 
> Line 88 in [192fb00](/fossasia/open-event-android/commit/192fb005917a0cbd74ddddcff797b7c6848140ae)
> 
>  if (it) {

But it is working fine it this condition because when edit profile is successful then only the value of isUpdated is set to **true** and let condition execute. And when failed, nothing happens. Why need to add extra if condition.

Reply:
but earlier you said
> But in that case when there is an error in updating the profile, then also it direct to profile fragment. Is that okay?

It is fine although if it is working correctly

Reply:
oh sorry, changes are done let me revert it.

Reply:
> but earlier you said
> 
> > But in that case when there is an error in updating the profile, then also it direct to profile fragment. Is that okay?
> 
> It is fine although if it is working correctly

That was different, In that case, no need to observe isUpdated. Directly on back pressed when toast message is observed.

Reply:
Updated.